[IV-24-25] Objetivo 3

Carlosmapego8 opened the pull request:
* [x] ¿Incluye el fichero `iv.yaml` las claves de todos los objetivos
anteriores?
* [x] ¿Se ha desarrollado el fichero para las tareas a partir de un
   issue?
* [x] ¿Ese issue está asignado a un milestone?
* [x] ¿El milestone incluye, entre sus productos mínimamente viables, la
  existencia de herramientas que permitan automatizar tareas?
* [x] ¿Hemos justificado correctamente en caso de elegir un gestor de
   dependencias el que se use como gestor de tareas?
* [x] ¿El PR, que también es un issue, está asignado al mismo milestone?
* [x] ¿La tarea `check` efectivamente comprueba la sintaxis de los fuentes
  existentes y sólo eso?
* [x] ¿Se ha comprobado que las instrucciones incluidas en el `README.md` son
completas y correctas?

Carlosmapego8 committed:
Elección gestor dependencias, #14

Carlosmapego8 committed:
Elección gestor tareas, #14

Carlosmapego8 committed:
Update README.md, closes #14

Carlosmapego8 committed:
Create justfile, closes #13

Carlosmapego8 committed:
Update iv.yaml, #13

Carlosmapego8 committed:
Update README.md

MarioRgzLpz commented:
Buenas se me ha asignado revisar esta PR. Te dejo algunas aclaraciones que sería conveniente hacer.

MarioRgzLpz commented on README.md:
```diff
@@ -34,3 +34,13 @@ Los contratos son de los siguientes tipos:
+- `just check`: verifica el correcto formateo del código.
+- `just clean`: limpia el binario y otros archivos.
+- `just build`: compila el proyecto y genera el binario.
+- `just installDeps`: instala las dependencias del proyecto.
```
Aqui tienes puesto `just installDeps` pero en el `justfile` has puesto `install_deps`.

Carlosmapego8 replied:
Se me coló al escribir, corregido en [f6f6426](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/f6f6426453e80be716ca7f4b6e413161755c02d9)

MarioRgzLpz commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,20 @@
+
+## Opciones consideradas
+
+- **Make**: Es una herramienta que es compatible con cualquier lenguaje, sin embargo, debido a esto no es un gestor específico de Go. En un principio, alguien acostumbrado a trabajar en Go no tiene por qué conocer cómo se utiliza, lo que hace que pueda conllevar bastante tiempo su aprendizaje.
+- **Sage**: Herramienta inspirada en Mage (un gestor de tareas ya existente escrito en Go)
+- **Task**: Es una herramienta que utiliza archivos en formato YAML y está pensada tanto para Go como para otros lenguajes.
+- **Just**: Es un gestor similar a Make pero con un enfoque en simplicidad y legibilidad. Está escrita en Go y se utiliza creando *Makefiles* que son mucho más similares a scripts.
```
Has dicho que hace cada herramienta, pero nada de si cumplen o no los criterios establecidos o de porque descartas cada una.

Carlosmapego8 replied:
Lo añado en [af8f02b](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/af8f02b0fcf0a6762c32047cc5afc5904c36d590)

MarioRgzLpz commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,20 @@
+## Elección
+Valorando los pros y contras de las opciones anteriormente expuestas, considero que la elección más sensata será el uso de Just. Su sencillez facilitará la automatización de las tareas repetitivas (compilación y ejecuciones) sumado a la facilidad futura de mantenimiento. Enlazando con la gestión de dependencias, al estar escrito en el propio lenguaje Go no necesitaremos tener en cuenta dependenci ...
+
+## Tareas a automatizar:
+- Compilación del proyecto y generación del binario en una carpeta `/bin`
+- Instalar las dependencias necesarias
+- Verificar el correcto formateado del código fuente 
```
Esto no es necesario que se encuentre aqui.

Carlosmapego8 replied:
Corregido en [f0519ab](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/f0519ab7a3820b9898ffbd497e52e82866c20dfe)

MarioRgzLpz commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,20 @@
+# Gestor tareas
+
+## Requisitos para la elección del gestor de tareas
+- Con el principal objetivo de reducir la deuda téncica en esta parte, la herramienta debe de ser una que esté actualizada.
+- Queremos continuar con las buenas prácticas que recomienda Go, así que elegiremos una herramienta que nos permita mantenerlas.
```
Aunque válidos criterios me parecen algo escasos. Considera añadir alguno más puesto que tu elección no se ha basado solo en estos dos.

Carlosmapego8 replied:
Añado más criterios en [af8f02b](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/af8f02b0fcf0a6762c32047cc5afc5904c36d590)

JJ replied:
¿Qué buenas prácticas, en concreto?

Carlosmapego8 replied:
Añado el enlace a las buenas practicas en [93611f9](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/93611f9262d97287d38096b836e5921f693320d6)

Carlosmapego8 committed:
Corregido error de escritura, #13

Carlosmapego8 committed:
Eliminación tareas a automatizar, #14

hossam1522 reviewed (commented):
Completamente de acuerdo con el compañero, destacar sobre todo la justificación de la elección, ya que los criterios son algo escasos y aún así en ningún momento estableces si las herramientas propuestas cumplen o no esos criterios.

Carlosmapego8 committed:
Clarificación elección gestor de tareas, #14

Carlosmapego8 commented:
Muchas gracias @MarioRgzLpz y @hossam1522 por vuestros consejos, ya he hecho algunos cambios para adaptarme a ellos.

Mario25402 commented on iv.yaml:
```diff
@@ -0,0 +1,5 @@
+lenguaje: Go
+entidad: internal/gestorturnos.go
+automatizar:
+  fichero: justfile.go
```
Una simple objeción, en el repositorio oficial de [just](https://github.com/casey/just) en el apartado de Quick Start del *README*, menciona que los fichero no son sensibles a que por ejemplo empiecen con mayúscula, incluso que tenga la extensión `.justifile` pero no se menciona de que `.go` sea una extensión correcta.

Carlosmapego8 replied:
Llevas razón, completamente de acuerdo contigo Mario. De hecho al fichero lo nombré correctamente, ha sido a la hora de incluirlo en el `iv.yaml` donde se me ha colado la errata. Ya lo he corregido en [f5a10f5](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/f5a10f5bd43d9d1d64a4802899c45e5d3f4aab90)

Carlosmapego8 committed:
Arreglo extensión del fichero, #13

hossam1522 commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,18 @@
+
+- **Make**: Es una herramienta que es compatible con cualquier lenguaje, sin embargo, debido a esto no es un gestor específico de Go, lo que hace que no fomente las mejores prácticas. En un principio, alguien acostumbrado a trabajar en Go no tiene por qué conocer cómo se utiliza, lo que hace que pueda conllevar bastante tiempo su aprendizaje. Al ser una herramienta altamente disponible no genera  ...
+- **Sage**: Herramienta inspirada en Mage (un gestor de tareas ya existente escrito en Go). Al estar inspirado en Mage ayudará a seguir las buenas prácticas de Go y será conocida por un desarrollador experimentado en Go. Sin embargo, Mage no recibe actualizaciones desde hace más de un año, lo que hace pensar que Sage pueda acabar por el mismo camino.
+- **Task**: Es una herramienta que utiliza archivos en formato YAML y está pensada tanto para Go como para otros lenguajes. Requiere un binario de la herramienta, lo que es una dependencia sumado al archivo YAML. A pesar de no estar escrito en lenguaje Go como otras herramientas, la sintáxis YAML es sencilla. 
+- **Just**: Es un gestor similar a Make pero con un enfoque en simplicidad y legibilidad. Está escrita en Go y se utiliza creando *Makefiles* que son mucho más similares a scripts. Al ser parecida a Make, su aprendizaje será rápido, pues junta su lógica con una sintáxis mucho más sencilla. Al estar escrita en Go permitirá continuar con las buenas prácticas y no sumar dependencias como otras herra ...
```
Aquí falla algo. Me refiero, si te vas a los repositorios puedes ver fácilmente que es al revés, Task está escrito en Go, mientras que Just está escrito en Rust.

Carlosmapego8 replied:
Arreglado en [60fb90b](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/60fb90b21c55aa0347e027ff070a3a5dd6cbcdb3)

Carlosmapego8 committed:
Cambios en opciones, #14

Carlosmapego8 commented:
@JJ revisión

JJ commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,18 @@
+## Opciones consideradas
+
+- **Make**: Es una herramienta que es compatible con cualquier lenguaje, sin embargo, debido a esto no es un gestor específico de Go, lo que hace que no fomente las mejores prácticas. En un principio, alguien acostumbrado a trabajar en Go no tiene por qué conocer cómo se utiliza, lo que hace que pueda conllevar bastante tiempo su aprendizaje. Al ser una herramienta altamente disponible no genera  ...
+- **Sage**: Herramienta inspirada en Mage (un gestor de tareas ya existente escrito en Go). Al estar inspirado en Mage ayudará a seguir las buenas prácticas de Go y será conocida por un desarrollador experimentado en Go. Sin embargo, Mage no recibe actualizaciones desde hace más de un año, lo que hace pensar que Sage pueda acabar por el mismo camino.
```
¿Ese gestor de tareas no merece estar en esta lista por qué razón?

Carlosmapego8 replied:
Añado Mage en [42f0f08](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/42f0f08e2e823c15d81450e123d0328cb9276d8a)

JJ commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,18 @@
+## Opciones consideradas
+
+- **Make**: Es una herramienta que es compatible con cualquier lenguaje, sin embargo, debido a esto no es un gestor específico de Go, lo que hace que no fomente las mejores prácticas. En un principio, alguien acostumbrado a trabajar en Go no tiene por qué conocer cómo se utiliza, lo que hace que pueda conllevar bastante tiempo su aprendizaje. Al ser una herramienta altamente disponible no genera  ...
+- **Sage**: Herramienta inspirada en Mage (un gestor de tareas ya existente escrito en Go). Al estar inspirado en Mage ayudará a seguir las buenas prácticas de Go y será conocida por un desarrollador experimentado en Go. Sin embargo, Mage no recibe actualizaciones desde hace más de un año, lo que hace pensar que Sage pueda acabar por el mismo camino.
```
¿Alguna razón por la que no enlaces la página que describa cada herramienta?

Carlosmapego8 replied:
Añado los enlaces en [42f0f08](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/42f0f08e2e823c15d81450e123d0328cb9276d8a)

JJ commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,18 @@
+## Requisitos para la elección del gestor de tareas
+- Con el principal objetivo de reducir la deuda téncica en esta parte, la herramienta debe de ser una que esté actualizada.
+- Queremos continuar con las buenas prácticas que recomienda Go, así que elegiremos una herramienta que nos permita mantenerlas.
+- Buscaremos una herramienta con una sintáxis sencilla de aprender y que no requiera de gran experiencia previa, de esta forma no gastaremos mucho tiempo en aprender a utilizarla si no estamos experimentados.
```
¿La sencillez es una medida objetiva?

Carlosmapego8 replied:
Dejo unos criterios algo más objetivos sobre la sencillez en [42f0f08](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/42f0f08e2e823c15d81450e123d0328cb9276d8a)

JJ commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,18 @@
+# Gestor tareas
+
+## Requisitos para la elección del gestor de tareas
+- Con el principal objetivo de reducir la deuda téncica en esta parte, la herramienta debe de ser una que esté actualizada.
```
¿Actualizada con qué frecuencia?

Carlosmapego8 replied:
Añade una frecuencia en la actualización además de una aclaración extra en [42f0f08](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/42f0f08e2e823c15d81450e123d0328cb9276d8a)

JJ reviewed (changes requested):
aparte de lo comentado, en los [errores frecuentes](https://github.com/JJ/IV-/blob/master/errores/objetivo-3.md) dice:

> En este objetivo, como en todos, se trata de que busquéis y trabajéis, no que copiéis lo que han hecho otros, lo que os diga la IA, o lo que salga en la primera página de Google. Si sólo "encuentras" make, sage, task y just, busca un poco más, porque hay muchas más herramientas.

Carlosmapego8 committed:
Cambios en gestores de tareas, #14

Se añaden nuevos gestores a tener en cuenta junto a la documentación oficial de todos ellos. Se cambian algunos requisitos para obtener unos criterios más objetivos

Carlosmapego8 commented:
@JJ ya he resuelto todos los comentarios, además de buscar nuevos gestores. Revisión

Carlosmapego8 committed:
Enlazadas buenas prácticas, #14

Carlosmapego8 commented:
@JJ Revisión

JJ commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,23 @@
+# Gestor tareas
+
+## Requisitos para la elección del gestor de tareas
+- Con el principal objetivo de reducir la deuda téncica en esta parte, la herramienta debe de ser una que esté actualizada. Consideraremos que la herramienta está actualizada si bien su última actualización fue hace menos de 6 meses.
```
¿Qué deuda hay y en qué cantidad la disminuye cada una de ellas?

Carlosmapego8 replied:
Corregido en [5cd7138](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/5cd71380056fa1eea1ad0d2a70ce983e78d7c33d)

JJ commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,23 @@
+
+## Requisitos para la elección del gestor de tareas
+- Con el principal objetivo de reducir la deuda téncica en esta parte, la herramienta debe de ser una que esté actualizada. Consideraremos que la herramienta está actualizada si bien su última actualización fue hace menos de 6 meses.
+- Queremos continuar con las [buenas prácticas que recomienda Go](https://go.dev/doc/effective_go), así que elegiremos una herramienta que nos permita mantenerlas.
```
¿Cuál de las buenas prácticas que has enlazado, en concreto?

Carlosmapego8 replied:
Corregido en [5cd7138](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/5cd71380056fa1eea1ad0d2a70ce983e78d7c33d)

JJ commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,23 @@
+## Requisitos para la elección del gestor de tareas
+- Con el principal objetivo de reducir la deuda téncica en esta parte, la herramienta debe de ser una que esté actualizada. Consideraremos que la herramienta está actualizada si bien su última actualización fue hace menos de 6 meses.
+- Queremos continuar con las [buenas prácticas que recomienda Go](https://go.dev/doc/effective_go), así que elegiremos una herramienta que nos permita mantenerlas.
+- Buscaremos una herramienta con la que, en caso de no tener experiencia previa, su uso no suponga un cuello de botella por la complejidad que suponga su aprendizaje (en este caso se tendrá en cuenta la calidad y cantidad de documentación desde la que se podrá llevar a cabo este aprendizaje). En este punto, evidentemente será preferible seleccionar una herramienta que previamente se conozca.
```
¿Esto es objetivo? ¿Es medible?

Carlosmapego8 replied:
Corregido en [5cd7138](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/5cd71380056fa1eea1ad0d2a70ce983e78d7c33d)

JJ commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,23 @@
+- Con el principal objetivo de reducir la deuda téncica en esta parte, la herramienta debe de ser una que esté actualizada. Consideraremos que la herramienta está actualizada si bien su última actualización fue hace menos de 6 meses.
+- Queremos continuar con las [buenas prácticas que recomienda Go](https://go.dev/doc/effective_go), así que elegiremos una herramienta que nos permita mantenerlas.
+- Buscaremos una herramienta con la que, en caso de no tener experiencia previa, su uso no suponga un cuello de botella por la complejidad que suponga su aprendizaje (en este caso se tendrá en cuenta la calidad y cantidad de documentación desde la que se podrá llevar a cabo este aprendizaje). En este punto, evidentemente será preferible seleccionar una herramienta que previamente se conozca.
+- Intentaremos que su uso genere la menor cantidad de dependencias posibles, de esta forma conseguiremos que la herramienta no sea un problema por ella misma.
```
¿De qué tipo de dependencias estamos hablando? ¿Cómo mides la "cantidad" de dependencias?

Carlosmapego8 replied:
Corregido en [5cd7138](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/5cd71380056fa1eea1ad0d2a70ce983e78d7c33d)

JJ commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,23 @@
+- Queremos continuar con las [buenas prácticas que recomienda Go](https://go.dev/doc/effective_go), así que elegiremos una herramienta que nos permita mantenerlas.
+- Buscaremos una herramienta con la que, en caso de no tener experiencia previa, su uso no suponga un cuello de botella por la complejidad que suponga su aprendizaje (en este caso se tendrá en cuenta la calidad y cantidad de documentación desde la que se podrá llevar a cabo este aprendizaje). En este punto, evidentemente será preferible seleccionar una herramienta que previamente se conozca.
+- Intentaremos que su uso genere la menor cantidad de dependencias posibles, de esta forma conseguiremos que la herramienta no sea un problema por ella misma.
+- Adecuarnos a la complejidad necesaria para el proyecto. Algunas herramientas pueden ser más sofisticadas pero requirir una configuración complicada desde el principio. Sabiendo que el proyecto que estamos tratando no es de gran tamaño, buscaremos alguna herramienta que pueda ser capaz de funcionar con la mínima configuración.
```
¿Cómo mides esta adecuación?

Carlosmapego8 replied:
Corregido en [5cd7138](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/5cd71380056fa1eea1ad0d2a70ce983e78d7c33d)

JJ reviewed (changes requested):
Aparte de repetir lo que dice en el [guión](https://jj.github.io/IV/documentos/proyecto/3.Automatizar) sobre las herramientas, que son recomendaciones generales y que por tanto, salvo que realmente los identifiques y los cuantifiques, no valen para que los repitas aquí (ya están en el guión, ¿qué aporta que lo pongas aquí?) te dice, dos veces:
- Criterios *objetivos*
- "Hay que evitar criterios subjetivos" y te cita explícitamente "familiaridad" (que es tu último criterio).

En [errores comunes](https://github.com/JJ/IV-/blob/master/errores/objetivo-3.md), por otro lado, menciona cosas como:

> En general, conviene que se eviten criterios que o no se sabe qué significan o son simples excusas para seleccionar uno sobre otro. En muchos casos no habrá diferencia entre diferentes gestores, simplemente diferentes formas de hacer las cosas y se tendrá que usar algún otro criterio adicional o simplemente elegir uno cualquiera.

Siento no haberte revisado antes, se me pasó.

Carlosmapego8 committed:
Cambios en requisitos, #14

Se readaptan los requisitos para buscas su objetividad

Carlosmapego8 commented:
@JJ revisión

JJ commented on docs/gestor_tareas.md:
```diff
@@ -1,11 +1,8 @@
-- Buscaremos una herramienta con la que, en caso de no tener experiencia previa, su uso no suponga un cuello de botella por la complejidad que suponga su aprendizaje (en este caso se tendrá en cuenta la calidad y cantidad de documentación desde la que se podrá llevar a cabo este aprendizaje). En este punto, evidentemente será preferible seleccionar una herramienta que previamente se conozca.
-- Intentaremos que su uso genere la menor cantidad de dependencias posibles, de esta forma conseguiremos que la herramienta no sea un problema por ella misma.
-- Adecuarnos a la complejidad necesaria para el proyecto. Algunas herramientas pueden ser más sofisticadas pero requirir una configuración complicada desde el principio. Sabiendo que el proyecto que estamos tratando no es de gran tamaño, buscaremos alguna herramienta que pueda ser capaz de funcionar con la mínima configuración.
+- Se busca que haya actualizaciones por lo menos en los últimos 12 meses para considerar que la herramienta está activa. Si esto no se cumple, podríamos considerar que hay una falta de mantenimiento, lo que conllevaría un aumento de la deuda técnica. En el caso de que esto no se cumpla, veremos si la herramienta es estable y contrastada (como por ejemplo lo es Make). En este caso, al haber alcanz ...
```
¿A qué comentario se debe que cambies "6 meses" a "12 meses"? 
Todo lo que hay después de "En el caso de que no se cumpla", ¿a qué comentario responde? Por otro lado, si es un criterio, ¿sería un criterio diferente?

Carlosmapego8 replied:
No es debido a ningún comentario, simplemente reflexionando consideré que una mayor ventana de tiempo podría ser algo más adecuado.
>¿A qué comentario se debe que cambies "6 meses" a "12 meses"?

No creo que sea otro criterio diferente, pues todo este criterio tiene relación con la frecuencia de actualización. Es un añadido del mismo criterio.
>Por otro lado, si es un criterio, ¿sería un criterio diferente?

JJ commented on docs/gestor_tareas.md:
```diff
@@ -1,11 +1,8 @@
-- Intentaremos que su uso genere la menor cantidad de dependencias posibles, de esta forma conseguiremos que la herramienta no sea un problema por ella misma.
-- Adecuarnos a la complejidad necesaria para el proyecto. Algunas herramientas pueden ser más sofisticadas pero requirir una configuración complicada desde el principio. Sabiendo que el proyecto que estamos tratando no es de gran tamaño, buscaremos alguna herramienta que pueda ser capaz de funcionar con la mínima configuración.
+- Se busca que haya actualizaciones por lo menos en los últimos 12 meses para considerar que la herramienta está activa. Si esto no se cumple, podríamos considerar que hay una falta de mantenimiento, lo que conllevaría un aumento de la deuda técnica. En el caso de que esto no se cumpla, veremos si la herramienta es estable y contrastada (como por ejemplo lo es Make). En este caso, al haber alcanz ...
+- Buscaremos una herramienta con la que, en caso de no tener experiencia previa, su uso no suponga un cuello de botella por la complejidad que suponga su aprendizaje. Para ello, su comunidad deber ser activa y amplia, esto también nos indicará el punto de madurez de la herramienta. Para medirlo, se utilizará estadísticas de su propio repositorio de github como el número de forks y estrellas que t ...
```
En [este comentario](https://github.com/Carlosmapego8/GoMountain/pull/16#discussion_r1888426460) preguntaba:

> https://github.com/Carlosmapego8/GoMountain/pull/16#discussion_r1888426460

Dices "corregido en [5cd7138](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/5cd71380056fa1eea1ad0d2a70ce983e78d7c33d)" pero lo que has hecho es borrarlo. ¿Ya no es de interés continuar con las buenas prácticas?

Por otro lado, ¿qué tiene que ver la "complejidad del aprendizaje" con "la comunidad activa y amplia"? ¿O cualquiera de las dos cosas con un criterio objetivo? En [este comentario](https://github.com/Carlosmapego8/GoMountain/pull/16#discussion_r1888426781) te preguntaba si esto era objetivo y medible. ¿Cambiar el párrafo de lugar lo hace objetivo y medible?

Carlosmapego8 replied:
Sigue siendo de interés, pero tras ver su pregunta formulada y mi posterior reflexión, consideré que no hay un criterio objetivo y medible para ver que gestores cumplen _mejor_ las buenas prácticas. Lo que me hizo borrarlo como criterio.
>Dices "corregido en [5cd7138](https://github.com/Carlosmapego8/GoMountain/pull/16/commits/5cd71380056fa1eea1ad0d2a70ce983e78d7c33d)" pero lo que has hecho es borrarlo. ¿Ya no es de interés continuar con las buenas prácticas?


Lo explico en el mismo criterio (creo), si no se conoce la herramienta y no hay una comunidad que te permita aprenderla y conocer cómo proceder será muy difícil aprenderla de ahí la relación. Por otro lado el no conocer la herramienta evidentemente es algo completamente subjetivo, sin embargo, el tamaño y actividad de la comunidad sí es objetivo y medible(de hecho aporto unas propuestas para su medición) de ahí el considerarlo como criterio. Podría considerarse que la parte de no tener experiencia previa es un pequeño preámbulo al criterio real que es el de la comunidad.
>Por otro lado, ¿qué tiene que ver la "complejidad del aprendizaje" con "la comunidad activa y amplia"? ¿O cualquiera de las dos cosas con un criterio objetivo?En https://github.com/Carlosmapego8/GoMountain/pull/16#discussion_r1888426781 te preguntaba si esto era objetivo y medible. ¿Cambiar el párrafo de lugar lo hace objetivo y medible?

JJ reviewed (changes requested):
El objeto de las correcciones es, a través de la reflexión sobre los comentarios que se hacen, llegar a una mejor comprensión de los conceptos de cada objetivo. Por eso, conviene que se consideren todos los comentarios de forma individual.

blancagiron commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,20 @@
+# Gestor tareas
+
+## Requisitos para la elección del gestor de tareas
+- Se busca que haya actualizaciones por lo menos en los últimos 12 meses para considerar que la herramienta está activa. Si esto no se cumple, podríamos considerar que hay una falta de mantenimiento, lo que conllevaría un aumento de la deuda técnica. En el caso de que esto no se cumpla, veremos si la herramienta es estable y contrastada (como por ejemplo lo es Make). En este caso, al haber alcanz ...
```
Hola, vengo a ver si puedo ayudarte un poco teniendo en cuenta comentarios que me hicieron a mi en su momento. En lo que se refiere a  " En este caso, al haber alcanzado un alto nivel de madurez (...)"¿Has pensado en que si un criterio no es necesario por algo, eso podría significar que es otro criterio? ¿Has considerado separar la frecuencia de actualizaciones de la estabilidad/madurez? Un gestor de tareas puede ser maduro y estable independientemente del ritmo al que se actualice.

juanmaaf replied:
Además del comentario de @blancagiron , te recomiendo simplificar un poco el criterio. Lo extiendes tantos y formulas tantos casos que dificulta la legibilidad y puede entenderse como totalmente subjetivo. Considera no hablar de las herramientas en la explicación del criterio. (No eliges Make en tu conclusión, pero esta redacción puede estar dando a entender de forma subjetiva que vas a tomar Make en consideración).

blancagiron commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,20 @@
+
+## Requisitos para la elección del gestor de tareas
+- Se busca que haya actualizaciones por lo menos en los últimos 12 meses para considerar que la herramienta está activa. Si esto no se cumple, podríamos considerar que hay una falta de mantenimiento, lo que conllevaría un aumento de la deuda técnica. En el caso de que esto no se cumpla, veremos si la herramienta es estable y contrastada (como por ejemplo lo es Make). En este caso, al haber alcanz ...
+- Buscaremos una herramienta con la que, en caso de no tener experiencia previa, su uso no suponga un cuello de botella por la complejidad que suponga su aprendizaje. Para ello, su comunidad deber ser activa y amplia, esto también nos indicará el punto de madurez de la herramienta. Para medirlo, se utilizará estadísticas de su propio repositorio de github como el número de forks y estrellas que t ...
```
> Por otro lado el no conocer la herramienta evidentemente es algo completamente subjetivo , sin embargo, el tamaño y actividad de la comunidad sí es objetivo y medible

En este caso, tú mismo has indicado esto. El hecho de no conocer la herramienta es subjetivo y por lo tanto se podría concluir que un criterio relacionado con la curva de aprendizaje no es objetivo ¿Has considerado no incluirlo? En lo que se refiere al tamaño y actividad de la comunidad ¿has considerado incluir el motivo de por qué este criterio es útil? es decir ¿para qué sirve que un gestor de tareas tenga una comunidad amplia? Y ya que mencionas la presencia en discusiones en foros, ¿has pensado la opción de incluir algún enlace de alguna discusión o de algún foro dónde las haya?

juanmaaf replied:
Te recomiendo revisar el PR de los compañeros que tengan público el repositorio y verás que se considera un error hablar de las estrellas del perfil de GitHub. Una herramienta puede tener un buen perfil de GitHub y no ser usada actualmente. Términos como "complejidad" y "curva de aprendizaje" son completamente subjetivos al no ser medibles e impiden que el criterio sea válido.

blancagiron reviewed (commented):
Hola, he hecho un par de comentarios relacionados con los criterios que a mí me ayudaron en su momento. Espero que te sirvan, ánimo

juanmaaf commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,20 @@
+- [**GoTaskR](https://github.com/Roemer/gotaskr): De nuevo otra herramienta escrita en Go que permite ejecutar tareas desde la línea de comandos. Al terminar su ejecución permite ver algunas estadísticas sobre lo ejecutado e incluso tiene algunas funcionalidades pensadas para DevOps específicamente. En la última actualización (hace escasos 2 meses) arreglaron algunos errores y sumado a todo esto  ...
+  
+## Elección
+Valorando los pros y contras de las opciones anteriormente expuestas, considero que la elección más sensata será el uso de Just. Su sencillez facilitará la automatización de las tareas repetitivas (compilación y ejecuciones) sumado a la facilidad futura de mantenimiento. Enlazando con la gestión de dependencias, a pesar de no estar escrita en Go no necesitaremos tener en cuenta dependencias extra ...
```
"Sencillez" y "facilidad futura" son conceptos totalmente subjetivos, ya que no son medibles. En este objetivo se deben establecer criterios medibles que permitan diferenciar las herramientas de forma objetiva.

juanmaaf replied:
Además, veo que también comparas las herramientas hablando de dependencias extra. No has especificado ese criterio, por lo que también es un error.

juanmaaf reviewed (commented):
Buenas, te dejo unos comentarios que espero que te sirvan de ayuda.
